### PR TITLE
[DO NOT MERGE] Update repo name references

### PR DESCRIPTION
**Pull request by colebemis**
**Do not merge until repo has been renamed to primer/react**

Updates repo references from `primer/components` to `primer/react`.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -10,19 +10,18 @@
...
-    <img src="https://img.shields.io/github/license/primer/components.svg" alt="">
+  <a aria-label="license" href="https://github.com/primer/react/blob/main/LICENSE">
+    <img src="https://img.shields.io/github/license/primer/react.svg" alt="">
   </a>
 </p>
 
-
 ## Documentation
 
 Our documentation site lives at [primer.style/components](https://primer.style/components). You'll be able to find detailed documentation on getting started, all of the components, our theme, our principles, and more.
```
Comment (bscofield) [inline README.md:27]: Should we investigate renaming this (down the road) as well?

**Review by bscofield — APPROVED** [review]
Looks good!

File: README.md (hunk 1 above)
Comment (smockle) [reply README.md:27]: Semi-related: Do we anticipate renaming the npm package from `@primer/components` → `@primer/react`?

Comment (colebemis) [reply README.md:27]: I think "yes" to both. 

Changing the website URL and package name will have bigger consequences so they'll need to be handled with more caution.

**Pull request closed without merging**